converts artists from according to our DB format

metavaults opened the pull request:
converts artists from MusicBrainz according to our DB format

metavaults committed:
converts artists from according to our DB format 

converts artists from MusicBrainz according to our DB format

kordianbruck commented on PutJSONTogether/app.js:
```diff
@@ -0,0 +1,70 @@
+function convertToInformation(content) {
+    fs.readFile(content, 'utf8', function readFileCallback(err, data){
+        if (err){
+            console.log(err);
```
If you add a return here, you can save on the else block and have an indentation less.

kordianbruck commented on PutJSONTogether/app.js:
```diff
@@ -0,0 +1,70 @@
+
+                    if (obj.table[i].aboutartist.hasOwnProperty("life-span")) {
+                        if(obj.table[i].aboutartist["life-span"].hasOwnProperty("begin")) {
+                            if (obj.table[i].aboutartist["life-span"].begin != null) {
```
You should probably introduce a local reference to `obj.table[i].aboutartist` to make this code more readable.

kordianbruck commented on PutJSONTogether/app.js:
```diff
@@ -0,0 +1,70 @@
+                                newObj.dateOfBirth = obj.table[i].aboutartist["life-span"].begin;
+                            } else { newObj.dateOfBirth = null; } 
+                        } else { newObj.dateOfBirth = null; }
+                    } else { newObj.dateOfBirth = null; }
```
Why don't you set the DoB null before the if, instead of having three else blocks?

kordianbruck commented on PutJSONTogether/app.js:
```diff
@@ -0,0 +1,70 @@
+            arrayofObj = obj.table.length;
+
+            for(i = 0; i < obj.table.length; i++) {
+                newObj = {};
```
Why don't you just initialize all the fields that we expect here with null instead of setting them below. Also a better name for the var would be great.

TimHenkelmann committed:
Implemented improvement proposals:
- add return to save on else block
- initialize fields as null
- introduce local reference to obj.table[i].aboutartist
- restructured if-statement regarding dateOfBirth and dateOfDeath (line30++)

TimHenkelmann commented:
hey @kordianbruck I just implemented your code proposals:

- add return to save on else block
- initialize fields as null
- introduce local reference to obj.table[i].aboutartist
- restructured if-statement regarding dateOfBirth and dateOfDeath (line30++)

could you have a look?
Thx!

kordianbruck commented on PutJSONTogether/app.js:
```diff
@@ -0,0 +1,62 @@
+                newObj.dateOfBirth = null;
+                newObj.dateOfDeath = null;
+                newObj.placeOfBirth = null;
+                newObj.placeOfDeath = null;
```
Please just move these to the object declaration. 
```javascript
newObj = {dateOfBirth: null, dateOfDeath: null,...};
```

kordianbruck commented on PutJSONTogether/app.js:
```diff
@@ -0,0 +1,62 @@
+var content = "artists.json";
+
+var musicians = [];
+var newObj = {};
```
This is not used anywhere

kordianbruck commented on PutJSONTogether/app.js:
```diff
@@ -0,0 +1,62 @@
+        arrayofObj = obj.table.length;
+
+        for (i = 0; i < obj.table.length; i++) {
+            newObj = {};
```
Move this inside the if, so the scope is clearer.

kordianbruck commented on PutJSONTogether/app.js:
```diff
@@ -0,0 +1,62 @@
+                newObj.psuedonym = null;
+                newObj.work = null;
+                newObj.release = null;
+                newObj.tag = null;
```
Move all of these lines up into the declaration as well please.

kordianbruck reviewed (commented):
Much better! 

Heres some more.

TimHenkelmann committed:
- removed unused newObj declaration
- moved "null" properties into object declaration

TimHenkelmann commented:
@kordianbruck next try!

kordianbruck reviewed (approved):
Perfect!
